internal/controller/gcpinstall/run.go

```diff
@@ -52,23 +70,85 @@ https://cloud.google.com/iam/docs/creating-managing-service-accounts`,
 	--display-name="${SERVICE_ACCOUNT}" \
 	--description="Runs Flightcrew's Control Tower VM."`,
 		}),
-		checkIAMRole,
-		command.NewWriteModel(command.Opts{
-			SkipIfSucceed: checkIAMRole,
-			Description:   "This command creates an IAM role from `${IAM_FILE}` for the Flightcrew VM to access configs and monitoring data.\n\nhttps://cloud.google.com/iam/docs/understanding-custom-roles",
-			Command: `gcloud iam roles create ${IAM_ROLE} \${PROJECT_OR_ORG_FLAG}
-	--file=${IAM_FILE}`,
-		}),
-		command.NewWriteModel(command.Opts{
-			SkipIfSucceed: checkIAMRole,
-			Description: `This command attaches the IAM role to Flightcrew's service account, which will give the IAM permissions to a new VM.
+	}
+}
+
+func getIAMRoleCommands(args map[string]string) []*command.Model {
+	newCheckIAMRole := func(replacer *strings.Replacer) *command.Model {
+		cmd := command.NewReadModel(command.Opts{
+			Description: "Check if a ${PERMISSIONS} Flightcrew IAM Role already exists or needs to be created.",
+			Command: `gcloud iam roles describe \${PROJECT_OR_ORG_FLAG}
+	"${ROLE}" >/dev/null 2>&1`,
+			Message: map[command.State]string{
+				command.PassState: "This Flightcrew IAM role already exists.",
+				command.FailState: "No IAM role found. Next step is to create one.",
```

Comment:
This can also fail if they don't have gcloud permissions to run command at all, in which case the next will fail as well. That's ok for now right?

Reply:
Hmmmmmmmm 🤔 I guess I should parse the error?